lib/beetle/queue_properties.rb

```diff
@@ -141,34 +128,37 @@ def retrieve_bindings(server, queue_name)
     end
 
     def remove_binding(server, queue_name, exchange, properties_key)
-      request_url = URI("http://#{server}/api/bindings/#{vhost}/e/#{exchange}/q/#{queue_name}/#{properties_key}")
-      request = Net::HTTP::Delete.new(request_url)
-
-      response = run_rabbit_http_request(request_url, request) do |http|
-        http.request(request)
-      end
+      response = run_api_request(server, Net::HTTP::Delete, "/api/bindings/#{vhost}/e/#{exchange}/q/#{queue_name}/#{properties_key}")
 
       unless %w(200 201 204).include?(response.code)
         log_error("Failed to remove obsolete binding for queue #{queue_name}", response)
         raise FailedRabbitRequest.new("Could not retrieve queue bindings")
       end
     end
 
-    def run_rabbit_http_request(uri, request, &block)
-      request.basic_auth(config.user, config.password)
-      case request.class::METHOD
-      when 'GET'
-        request["Accept"] = "application/json"
-      when 'PUT'
-        request["Content-Type"] = "application/json"
+    def run_api_request(server, request_const, path, *request_args)
```

Comment:
This change is bigger than we would have liked but I didn't find a good way to add this without making the code very hard to understand. Even for the short time we keep this running it still was not worth it in my estimation.

Note that this is, to a larger degree, due to me not knowing the code very well and maybe there was a simpler way, but at the same time, I don't think this change is bad.